### ISSUE TITLE
ShouldBeEquivalentTo() fails when a property throws an exception

### ISSUE THREAD

**Issue by sunsided** [dennisdoomen/FluentAssertions#362]
**TL;DR**: How can I make sure that `ShouldBeEquivalentTo()` ignores exceptions and/or skips certain fields of types?

I have a type `Maybe<T>` that basically acts as a wrapper around values that might or might not be set (sort of like `Nullable<T>`) and which has two relevant properties, `bool HasValue {get;}` and `T Value {get;}`. The logic ensures that if you call `Value` while `HasValue` is `false`, you'll end up with an `InvalidOperationException` being thrown.

Now that type is being used in a class, as, say `class Bar { public Maybe<string> Foo {get; set;} }`. When I call `bar.ShouldBeEquivalentTo(otherBar)` and one of the maybes doesn't have a value, the call fails due to that `InvalidOperationException` on accessing `Value` even if that behavior is expected.
As a result of the failed call, the unit test is (incorrectly) marked as failed.

Is there a way to suppress that?

**Comment by dennisdoomen** [comment]
You have several options:
* You can build a custom `IEquivalencyStep` and add those to the defaults (using `AssertionOptions`) or use it in each call to `ShouldBeEquivalentTo`.
* You could call the `Excluding` option that gives you a `ISubjectInfo` so that you can filter it out. 
* You can use the `Using<Maybe<TSomeType>>` option to change the behavior

**Comment by dennisdoomen** [comment]
@sunsided does this answer your question?

**Comment by sunsided** [comment]
@dennisdoomen Whoops sorry, totally forgot. Yes, it does indeed! Thanks!
